### DISCUSSION TITLE
Make strip == False work for binaries

### DISCUSSION THREAD

**Discussion by matanl99** [srand/jolt#67] [Ideas]
Hi!

I would like to make it possible to leave the symbols in the binary when strip == False. This is useful when using tools like ASAN to print the line numbers in its output. 

I think this would be a useful feature, do you agree?

If so, I would like to remove the highlighted line below when strip == "False" but I cannot find a way of accessing strip in this context. Do you have any ideas or suggestions on how to get access to "strip" here? Or should I do it some other way?

ninja.py line 1132
    linker = GNULinker(
        command=" && ".join([
            "$ld $ldflags $imported_ldflags $extra_ldflags $libpaths -Wl,--start-group @objects.list -Wl,--end-group -o $out -Wl,--start-group $libraries -Wl,--end-group",
            "$objcopy_path --only-keep-debug $out .debug/$binary",
            "$objcopy_path --strip-all $out",   <<<<<< remove this line when strip == "False"
            "$objcopy_path --add-gnu-debuglink=.debug/$binary $out"
        ]),

Regards

**Comment by srand** [comment]
Hey!

I would expect it to work out of the box already, but I haven't tested it. Have you added -g to cxxflags? Or are you hit by this https://gcc.gnu.org/bugzilla/show_bug.cgi?id=77631 perhaps? 

If so, I guess one option is to introduce another class attribute to control whether the debug info is separated into another file or kept in the executable. Not quite sure how I would go about implementing that, but I will give it some thought.

**Comment by matanl99** [comment]
Thank you !!! Yes you are right it does work!

I first tried put back the debug info by using "objcopy --add-gnu-debuglink=build/ninja-example_debug_true/.debug/example bin/example" but that gave me 
objcopy: bin/stR3Hb4a: debuglink section already exists
Maybe this can work but I don't know how to do that. Maybe the same bug you referenced.

Then I used eu-unstrip from elfutils package and managed to merge the two files together. Problem solved! :)
